paper/code_release/utils.py

```diff
@@ -0,0 +1,247 @@
+import dataclass
+import os
+from contextlib import contextmanager, redirect_stderr, redirect_stdout
+from os import devnull
+from pathlib import Path
+import struct
+from typing import Dict, List, Optional, Union
+from warnings import warn
+
+from openbabel import pybel
+import pandas as pd
+import molgrid
+import numpy as np
+from gridData import Grid
+from pkg_resources import resource_filename
+
+
+gninatypes = list(molgrid.GninaIndexTyper().get_type_names())
+
+""" Gninatypes from codebase:
+['Hydrogen',
+ 'PolarHydrogen',
+ 'AliphaticCarbonXSHydrophobe',
+ 'AliphaticCarbonXSNonHydrophobe',
+ 'AromaticCarbonXSHydrophobe',
+ 'AromaticCarbonXSNonHydrophobe',
+ 'Nitrogen',
+ 'NitrogenXSDonor',
+ 'NitrogenXSDonorAcceptor',
+ 'NitrogenXSAcceptor',
+ 'Oxygen',
+ 'OxygenXSDonor',
+ 'OxygenXSDonorAcceptor',
+ 'OxygenXSAcceptor',
+ 'Sulfur',
+ 'SulfurAcceptor',
+ 'Phosphorus',
+ 'Fluorine',
+ 'Chlorine',
+ 'Bromine',
+ 'Iodine',
+ 'Magnesium',
+ 'Manganese',
+ 'Zinc',
+ 'Calcium',
+ 'Iron',
+ 'GenericMetal',
+ 'Boron']
+"""
+
+gninatype_to_atom_num = {
+    "Hydrogen": 1,
+    "PolarHydrogen": 1,
+    "AliphaticCarbonXSHydrophobe": 6,
+    "AliphaticCarbonXSNonHydrophobe": 6,
+    "AromaticCarbonXSHydrophobe": 6,
+    "AromaticCarbonXSNonHydrophobe": 6,
+    "Nitrogen": 7,
+    "NitrogenXSDonor": 7,
+    "NitrogenXSDonorAcceptor": 7,
+    "NitrogenXSAcceptor": 7,
+    "Oxygen": 8,
+    "OxygenXSDonor": 8,
+    "OxygenXSDonorAcceptor": 8,
+    "OxygenXSAcceptor": 8,
+    "Sulfur": 16,
+    "SulfurAcceptor": 16,
+    "Phosphorus": 15,
+    "Fluorine": 9,
+    "Chlorine": 17,
+    "Bromine": 35,
+    "Iodine": 53,
+    "Magnesium": 12,
+    "Manganese": 25,
+    "Zinc": 30,
+    "Calcium": 20,
+    "Iron": 26,
+    "Boron": 6,
+    "GenericMetal": 0,
+}
+
+warn(
+    "Hardcoding atom types. Please only use this if the ligand-based grid is suppossed to operate at the element level. This will fail if we have i.e. aromatic and aliphatic carbons on different channels."
+)
+atom_num_to_gninatype_name = {
```

Comment:
Would be better to make the constants in this file uppercase

Reply:
I suggest not touching the code for now, as it is research code